src/DeferredImageStorageFilesystem.php

```diff
@@ -200,6 +200,12 @@ private function getConfigPath(string $path): string
      */
     private function decode(string $contents): array
     {
-        return json_decode($contents, true);
+        $content = json_decode($contents, true);
+
+        if (!\is_array($content)) {
+            throw new \RuntimeException('Invalid JSON: '.json_last_error_msg());
+        }
```

Comment:
```suggestion
        if (JSON_ERROR_NONE !== json_last_error()) {
            throw new \JsonException(json_last_error_msg());
        }
        
        if (!\is_array($content)) {
            throw new \InvalidArgumentException('Invalid JSON data: expected array, got "%s"', gettype($content)));
        }
```

How about this?

Reply:
`JsonException` is only available from PHP 7.4 😉

Reply:
🙈🙃
We could use https://github.com/symfony/polyfill/blob/master/src/Php73/Resources/stubs/JsonException.php ?

Reply:
Yes we could, but why would we need different exceptions at all? 🤷‍♂️

Reply:
We are using the polyfill already since #63